viewing_party/party.py

```diff
@@ -0,0 +1,264 @@
+    """
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    friends_titles = []
+    for friend in friends_watched:
+        for title in friend["watched"]:
+            friends_titles.append(title['title'])
+
+    unique_watched = []
+
+    for title in user_watched:
+        if title["title"] not in friends_titles:
+            unique_watched.append(title)
+
+    return unique_watched
+    
+    
+def get_friends_unique_watched(user_data):
+    """
+    Takes in one parameter user_data,
+    returns list of dictionaries friends_unique_watched,
+    representing movies at least one friend has watched,
+    but the user has not watched. 
+    """
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    
+    user_movies = [movie["title"] for movie in user_watched]
+    friends_unique_watched = []
+
+    for friend in friends_watched:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_movies:
+                if movie not in friends_unique_watched:
+                    friends_unique_watched.append(movie)
+
+    return friends_unique_watched
+
+
+# Wave 4
+def user_has_watched(movie, watched_movies):
+    """
+    Takes two parameters, movie and watched_movies.
+    Determines if user has watched a movie,
+    returns a Boolean. 
+    """
+    if movie in watched_movies:
+        return True
+    else:
+        return False
+
+
+def in_friends_watchlist(movie, watched_by_friends):
+    """
+    Takes two parameters, movie and watched_by_friends.
+    Determines whether at least one friend has watched the movie.
+    Returns a Boolean.
+    """
+    if movie in watched_by_friends:
+        return True
+    else:
+        return False
+
+
+def in_users_subscriptions(host, subscriptions):
+    """
+    Takes two parameters, host and subscriptions.
+    Determines whether the movie host is a service subscription.
+    Returns a Boolean.
+    """
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+
+def get_friends_movies(user_data):
+    """
+    Takes in one parameter user_data.
+    Determines which movies friends has watched.
+    Returns list of dictionaries, movies.
+    """
+    titles = set()
+    friends = user_data["friends"]
+    movies = []
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in titles:
+                movies.append(movie)
+                titles.add(movie["title"])
+    return movies
+
+
+def get_available_recs(user_data):
+    """
+    Takes in one parameter user_data.
+    Determines movies user has not watched, at least
+    one friend has watche, and is available to stream.
+    Returns list of dictionaries available_recs.
+    """
+    user_watched = user_data["watched"]
+    friends_movies = get_friends_movies(user_data)
+    available_recs = []
+
+    if len(friends_movies) == 0:
+        return []
+
```

Comment:
since available_recs starts out as an empty list, we can forgo this if statement, because on line 215 you are returning available_recs regardless!